contracts/GnosisSafe.sol

```diff
@@ -170,6 +170,8 @@ contract GnosisSafe
             // We only substract 2500 (compared to the 3000 before) to ensure that the amount passed is still higher than safeTxGas
             success = execute(to, value, data, operation, gasPrice == 0 ? (gasleft() - 2500) : safeTxGas);
             gasUsed = gasUsed.sub(gasleft());
+            // If no safeTxGas and no gasPrice was set (e.g. both are 0), then the internal tx is required to be successful
```

Comment:
Maybe it can be added to the comment that this behaviour is implemented due to the optimistic estimation of ethereum providers